File: wear/src/main/java/at/shockbytes/corey/fragment/WorkoutOverviewFragment.java
package at.shockbytes.corey.fragment;


import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.ActivityOptionsCompat;
import android.support.wearable.view.WearableRecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import at.shockbytes.corey.R;
import at.shockbytes.corey.adapter.WearWorkoutOverviewAdapter;
import at.shockbytes.corey.common.core.util.WorkoutNameComparator;
import at.shockbytes.corey.common.core.workout.model.Workout;
import at.shockbytes.corey.core.MainActivity;
import at.shockbytes.corey.core.WorkoutActivity;
import at.shockbytes.corey.util.MyOffsettingHelper;
import at.shockbytes.util.adapter.BaseAdapter;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * A simple {@link Fragment} subclass.
 */
public class WorkoutOverviewFragment extends Fragment
        implements BaseAdapter.OnItemClickListener<Workout>, MainActivity.OnWorkoutsLoadedListener {

    private static final String ARG_WORKOUTS = "arg_workouts";

    public static WorkoutOverviewFragment newInstance(ArrayList<Workout> workouts) {
        WorkoutOverviewFragment fragment = new WorkoutOverviewFragment();
        Bundle args = new Bundle();
        args.putParcelableArrayList(ARG_WORKOUTS, workouts);
        fragment.setArguments(args);
        return fragment;
    }


    @BindView(R.id.fragment_workout_overview_rv)
    protected WearableRecyclerView recyclerView;

    private List<Workout> workouts;

    private Unbinder unbinder;

    public WorkoutOverviewFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        workouts = getArguments().getParcelableArrayList(ARG_WORKOUTS);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v =  inflater.inflate(R.layout.fragment_workout_overview, container, false);
        unbinder = ButterKnife.bind(this, v);
        return v;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setupRecyclerView();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();

        if (unbinder != null) {
            unbinder.unbind();
        }
    }

    @Override
    public void onItemClick(Workout workout, View v) {

        startActivity(WorkoutActivity.newIntent(getContext(), workout),
                ActivityOptionsCompat.makeSceneTransitionAnimation(getActivity()).toBundle());
    }


    private void setupRecyclerView() {

        Collections.sort(workouts, new WorkoutNameComparator());
        WearWorkoutOverviewAdapter adapter = new WearWorkoutOverviewAdapter(getContext(), workouts);

        recyclerView.setCenterEdgeItems(true);
        recyclerView.setOffsettingHelper(new MyOffsettingHelper());
        recyclerView.setAdapter(adapter);
        adapter.setOnItemClickListener(this);
    }

    @Override
    public void onWorkoutLoaded(List<Workout> workouts) {

        this.workouts = new ArrayList<>(workouts);
        setupRecyclerView();
    }
}

File: wear/src/main/java/at/shockbytes/corey/ui/fragment/WorkoutOverviewFragment.kt
package at.shockbytes.corey.ui.fragment

import android.os.Bundle
import androidx.core.app.ActivityOptionsCompat
import androidx.wear.widget.WearableLinearLayoutManager
import androidx.wear.widget.WearableRecyclerView
import android.view.View
import at.shockbytes.corey.R
import at.shockbytes.corey.ui.adapter.WearWorkoutOverviewAdapter
import at.shockbytes.corey.common.core.util.WorkoutNameComparator
import at.shockbytes.corey.common.core.workout.model.Workout
import at.shockbytes.corey.dagger.WearAppComponent
import at.shockbytes.corey.ui.activity.WearMainActivity
import at.shockbytes.corey.ui.activity.WorkoutActivity
import at.shockbytes.util.adapter.BaseAdapter
import kotterknife.bindView

class WorkoutOverviewFragment : WearableBaseFragment(),
        BaseAdapter.OnItemClickListener<Workout>, WearMainActivity.OnWorkoutsLoadedListener {

    private lateinit var workouts: List<Workout>

    private val recyclerView: WearableRecyclerView by bindView(R.id.fragment_workout_overview_rv)

    override val layoutId = R.layout.fragment_workout_overview

    override fun setupViews() {
        val adapter = WearWorkoutOverviewAdapter(requireContext(), this).apply {
            data = workouts.sortedWith(WorkoutNameComparator()).toMutableList()
        }

        recyclerView.isEdgeItemsCenteringEnabled = true
        recyclerView.layoutManager = WearableLinearLayoutManager(context)
        recyclerView.adapter = adapter
    }

    override fun injectToGraph(appComponent: WearAppComponent) = Unit

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        workouts = arguments?.getParcelableArrayList(ARG_WORKOUTS)
                ?: throw NullPointerException("Workout must be set!")
    }

    override fun bindViewModel() {
    }

    override fun onItemClick(t: Workout, position: Int, v: View) {
        activity?.let {
            startActivity(WorkoutActivity.newIntent(it, t),
                    ActivityOptionsCompat.makeSceneTransitionAnimation(it).toBundle())
        }
    }

    override fun onWorkoutLoaded(workouts: List<Workout>) {
        this.workouts = workouts
        setupViews()
    }

    companion object {

        private const val ARG_WORKOUTS = "arg_workouts"

        fun newInstance(workouts: ArrayList<Workout>): WorkoutOverviewFragment {
            val fragment = WorkoutOverviewFragment()
            val args = Bundle()
            args.putParcelableArrayList(ARG_WORKOUTS, workouts)
            fragment.arguments = args
            return fragment
        }
    }
}
